pkg/util/kubernetes/apiserver/metadata_controller.go

```diff
@@ -245,16 +246,13 @@ func (m *MetadataController) mapEndpoints(endpoints *corev1.Endpoints) error {
 	svc := endpoints.Name
 	namespace := endpoints.Namespace
 	for nodeName, ns := range nodeToPods {
-		metaBundle := m.store.getOrCreate(nodeName)
-
-		metaBundle.m.Lock()
+		metaBundle := m.store.getCopyOrNew(nodeName)
 		metaBundle.Services.Delete(namespace, svc) // cleanup pods deleted from the service
```

Comment:
this won't affect the one in the store though ? 🤔

Reply:
it does, we do a `store.set(...)` https://github.com/DataDog/datadog-agent/pull/3404/files/2c04b9d3176a63bef33ed50508ca5dd948bcacf6#diff-88ac0183412502946693e4490295addaL257